EasyPost/Models/API/Order.cs

```diff
@@ -0,0 +1,130 @@
+        public CustomsInfo customs_info { get; set; }
+        [JsonProperty("from_address")]
+        public Address from_address { get; set; }
+        [JsonProperty("is_return")]
+        public bool? is_return { get; set; }
+        [JsonProperty("messages")]
+        public List<Message> messages { get; set; }
+        [JsonProperty("rates")]
+        public List<Rate> rates { get; set; }
+        [JsonProperty("reference")]
+        public string reference { get; set; }
+        [JsonProperty("return_address")]
+        public Address return_address { get; set; }
+        [JsonProperty("service")]
+        public string service { get; set; }
+        [JsonProperty("shipments")]
+        public List<Shipment> shipments { get; set; }
+        [JsonProperty("to_address")]
+        public Address to_address { get; set; }
+
+        #endregion
+
+        #region CRUD Operations
+
+        /// <summary>
+        ///     Purchase the shipments within this order with a carrier and service.
+        /// </summary>
+        /// <param name="withCarrier">The carrier to purchase a shipment from.</param>
+        /// <param name="withService">The service to purchase.</param>
+        [CrudOperations.Update]
+        public async Task<Order> Buy(string withCarrier, string withService)
+        {
+            if (id == null)
+            {
+                throw new Exception("id is null");
+            }
+
+            Dictionary<string, object> parameters = new Dictionary<string, object>
+            {
+                {
+                    "carrier", withCarrier
+                },
+                {
+                    "service", withService
+                }
+            };
+
+            return await Update<Order>(Method.Post, $"orders/{id}/buy", parameters);
+        }
+
+        /// <summary>
+        ///     Purchase a label for this shipment with the given rate.
+        /// </summary>
+        /// <param name="rate">EasyPost.Rate object instance to purchase the shipment with.</param>
+        [CrudOperations.Update]
+        public async Task Buy(Rate rate)
+        {
+            if (rate.carrier != null)
+            {
+                if (rate.service != null)
+                {
+                    await Buy(rate.carrier, rate.service);
+                }
+                else
+                {
+                    throw new Exception("service is required");
+                }
+            }
+            else
+            {
+                throw new Exception("carrier is required");
+            }
+        }
+
+        /// <summary>
+        ///     Populate the rates property for this Order.
+        /// </summary>
+        [CrudOperations.Update]
+        public async Task GetRates()
+        {
+            if (id == null)
+            {
+                throw new Exception("id is null");
+            }
+
+            Order order = await Request<Order>(Method.Get, $"orders/{id}/rates");
+            rates = order.rates;
```

Comment:
We should `Merge` function here to update the current object

Reply:
Resolved.